### PR TITLE
UEV+ Material Diversity

### PR DESCRIPTION
**Pull request by NotAPenguin0**
## Motivation

UEV+ is still lacking a lot of content as it is right now, and I believe part of this is due to a lack of materials. Essentially every supposedly expensive material is infinity, spacetime or something derived from it. This PR attempts to address the overuse of infinity by introducing a new material: *Proto-Halkonite Steel*. This material is initially made out of infinity, but later on more efficient recipes are unlocked that use advanced godforge materials. With that said, let's go into all the changes I made in this PR. Some of these recipe changes are in the corresponding coremod PR instead.

## Tier 8 Waterline 

Currently, tier 8 waterline is fully optional, though it will be required by the godforge when that is finished. I want to make it required slightly earlier in progression, but for that to happen its recipes need to be buffed as they are far too expensive to be reasonable in UEV.

- Empty Quark Catalyst Housing: 4 UEV Field Generator each -> 1 UEV Field Generator each. 
afbeelding
- Particle Beam Guidance Casing: 1 UEV Field Generator per 4 -> 1 UEV Pump per 4
afbeelding
- Femtometer-Calibrated Particle Beam Casing: 1 UEV Field Generator per 4 -> 1 UHV Field Generator per 4
afbeelding
- Quark Exclusion Casing: All inputs except UEV Field Generator doubled, output doubled from 4 to 8.
afbeelding

Additionally, the Baryonic Perfection process is adjusted to produce an excess *500L Stabilized Baryonic Matter* per parallel for each cycle. This makes it a useful material for processing outside the waterline, though with relatively scarce supply. Producing it costs 1 UEV amp for every 500L/120s.

## Dimensionally Shifted Superfluid

The fact that UIV+ components still use regular lubricant from drilled oil is odd, so this new material addresses that. All lubricant in electric components is replaced with an equivalent amount of Dimensionally Shifted Superfluid instead.
afbeelding

This fluid is produced in the DTPF, with an initial recipe at Awakened Draconium Coils, and a better one unlocking with Hypogen Coils.
afbeelding
*Stabilized Baryonic Matter, Metastable Oganesson, Subatomically Perfect Water, Celestial Tungsten Plasma and Super Heavy Radox make Dimensionally Shifted Superfluid*
afbeelding
*Faster, identical recipe to the early one, but Super Heavy Radox is replaced with Heavy Radox, which can be produced by the QFT*.

## Proto-Halkonite Steel

Proto-Halkonite Steel is a new material that **fully replaces the use of Fine Infinity Wire in UIV motors and field generators**. Proto-Halkonite Steel is completely unbreakable: Once forged into a part (ingot, wire, plate), it cannot be changed in any way. As the name suggests, this is only an early form of the material. Exo-Halkonite Steel is also added, but currently unimplemented aside from textures. The implementation of Exo-Halkonite Steel will come with exotic circuits.

afbeelding
*Molten Proto-Halkonite Steel Base recipe.*

afbeelding
*Creating Hot Proto-Halkonite Steel Parts*

afbeelding
*Proto-Halkonite Steel Cooling*

afbeelding
*Use of Proto-Halkonite Steel in UIV components*

Proto-Halkonite Steel is produced by pouring Molten Proto-Halkonite Steel Base Compound over a finished part (rod, plate, ingot). Initially, it's only possible to do this using infinity parts, meaning UIV components will cost 12.5% more infinity than they do currently. However, with a UIV energy hatch and Transcendent Plasma Mixer, alternative recipes are unlocked that use Creon or Mellion parts instead, significantly cutting down the infinity cost and processing time.

Because Proto-Halkonite Steel cannot be altered, it cannot be used in molten form in CoAL recipes. To fix this, Superdense Proto-Halkonite Steel Plates are added. These plates are worth 64 ingots of Proto-Halkonite Steel, and can only be made in the Black Hole Compressor. The CoAL recipe generator is adjusted to use these superdense plates instead of molten Proto-Halkonite Steel.

Credit for Halkonite Steel goes to @IncandescenceSy

## Plasma Mixer Downtier

Introducing these changes adds significantly more DTPF usage before UIV components can be crafted, so the Transcendent Plasma Mixer is downtiered accordingly to use UEV components instead of UIV components, and Proto-Halkonite Steel instead of Transcendent Metal. This should place it slightly earlier in progression and ease the burden on DTPF catalyst production spam. 
afbeelding

## Alternate Metastable Oganesson Recipe

afbeelding
Adds an alternative recipe for Metastable Oganesson in the Mk5 fusion reactor, using copper plasma that can only be obtained from the godforge plasma module.

## Deep Dark Downtier

Currently the Deep Dark portal comes way too late in progression to be truly useful. Yes you get tritanium out of it, but that's about the only useful thing (and fusing it at that point is faster than smelting anyway). The only reason you truly make the portal is because you have to to progress. This PR adjusts the components used in the DD portal recipe from UMV to UIV. This makes it more useful, and additionally allows players to access a better source of tartarite, an important input in Proto-Halkonite Steel.

## Overall implications on UEV/UIV progression

To more easily visualize all these changes, I compiled a crude flowchart that shows how all these new materials and changes interact with each other.
afbeelding

### CODE REVIEW

**Comment by NotAPenguin0** [comment]
In terms of content, this PR is pretty much finished. It's only waiting for texture work.

**Comment by Connor-Colenso** [comment]
Sounds good. These tiers certainly need more content to flesh them out but seems like good strides are being made towards that. 👍🏻

**Comment by boubou19** [comment]
making a main material per tier allows for less boring recipes, and it can help reduce spam. I'm good with this change.

**Comment by NotAPenguin0** [comment]
Assets are mostly finished (thank you @IncandescenceSy), this is now just waiting on #3050

File: src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java [hunk 1]
```diff
@@ -300,6 +301,8 @@ private static FluidStack tryConvertItemStackToFluidMaterial(ItemStack input) {
             if (strippedOreDict.contains("PTMEG")) return FluidRegistry.getFluidStack(
                 "molten.silicone",
                 (int) (orePrefix.mMaterialAmount / (GTValues.M / 144)) * input.stackSize);
+            if (strippedOreDict.contains("PTMEG")) return FluidRegistry
+                .getFluidStack("molten.silicone", (int) (orePrefix.mMaterialAmount / (M / 144)) * input.stackSize);
```
Comment (serenibyss) [inline src/main/java/goodgenerator/loader/ComponentAssemblyLineRecipeLoader.java:305]: Is this intentionally added?

File: src/main/java/gregtech/loaders/materials/MaterialsInit1.java [hunk 2]
```diff
@@ -902,7 +902,7 @@ public static void load() {
...
-        Materials.Bedrockium              = new MaterialBuilder(395,TextureSet.SET_DULL, "Bedrockium").addOreItems().addDustItems().addMetalItems().setDurability(327680).setToolSpeed(8f).setToolQuality(9).setRGB(50,50,50).setName("Bedrockium").setBlastFurnaceRequired(true).setBlastFurnaceTemp(9900).setMeltingPoint(9900).setColor(Dyes.dyeBlack).setOreValue(4).setDensityDivider(1).setDensityMultiplier(1).constructMaterial().setProcessingMaterialTierEU(TierEU.RECIPE_EV).disableAutoGeneratedVacuumFreezerRecipe();
+        Materials.Bedrockium              = new MaterialBuilder(395,TextureSet.SET_DULL, "Bedrockium").addOreItems().addDustItems().addMetalItems().setDurability(327680).setToolSpeed(8f).setToolQuality(9).setRGB(50,50,50).setName("Bedrockium").setBlastFurnaceRequired(true).setBlastFurnaceTemp(9900).setMeltingPoint(9900).setMaterialList(new MaterialStack(Materials.SiliconDioxide, 26244), new MaterialStack(Materials.Diamond, 9)).setColor(Dyes.dyeBlack).setOreValue(4).setDensityDivider(1).setDensityMultiplier(1).constructMaterial().setProcessingMaterialTierEU(TierEU.RECIPE_EV).disableAutoGeneratedVacuumFreezerRecipe();
```
Comment (serenibyss) [inline src/main/java/gregtech/loaders/materials/MaterialsInit1.java:905]: Will this now have a decomposition recipe autogenerated?

File: src/main/java/gregtech/common/items/IDMetaItem03.java [hunk 3]
```diff
@@ -232,7 +232,7 @@ public enum IDMetaItem03 {
     Transdimensional_Alignment_Matrix(758),
     Thermal_Superconductor(759),
     Relativistic_Heat_Capacitor(760),
-    Phononic_Seed_Crystal(761);
+    Phononic_Seed_Crystal(761),;
```
Comment (serenibyss) [inline src/main/java/gregtech/common/items/IDMetaItem03.java:235]: Unnecessary change

**Review by serenibyss — COMMENTED** [review]
Overall looks good, I have a few code comments and a few suggestions for more places to use Dimensionally Shifted Superfluid over normal Lubricant:
- Assembly Line:
    - T3 field restriction coil (maybe? mostly uev but uses transcendent metal)
    - T4 field restriction coil
    - UIV/UMV/UXV CoAL casing
    - Space Elevator Motor MK3 (maybe? similar to T3 field restriction coil)
    - Space Elevator Motor MK4+5
    - Dyson Deployment Base Casing + Unit Core
    - Fluid Cell Block T9+T10 (yottank)

File: src/main/java/gregtech/loaders/materials/MaterialsInit1.java (hunk 2 above)
Comment (NotAPenguin0) [reply src/main/java/gregtech/loaders/materials/MaterialsInit1.java:905]: Sadly not

**Comment by serenibyss** [comment]
Handful of suggestions for places to use proto-halkonite, feel free to use or ignore any of these (I may suggest multiple options for different recipes just to give some diversity, unless specified I usually mean to only pick one if any instead of both):
- TFFT Tier IX block
    - 2x6 dense transcendent metal plate -> 2x6 dense proto-halkonite steel plate
    - 6 hypogen rotor -> 6 proto-halkonite steel rotor
- T3 qft shielding core
    - 16 dense transcendent metal plate -> 16 dense proto-halkonite steel plate
- Space Pumping Module MK2 (would need both)
    - (AL) 2304L molten transcendent metal -> 2304L proto-halkonite preparation base
    - (Assembler) 16 double transcendent metal plate -> 16 double proto-halkonite steel plate
- Space Elevator Motor MK4
    - 16 Osmiridium Plate -> 16 proto-halkonite steel plate
- Space Assembler Module MK2
    - 8 transcendent metal gear -> 8 proto-halkonite steel gear
    - 16 small transcendent metal gear -> 16 small proto-halkonite steel gear
- UIV CoAL casing:
    - Replace one or two of the 3 transcendent metal parts with proto-halkonite steel
- Fluid Cell Block T9/T10
    - 12 double awakened draconium plate -> 12 double proto-halkonite steel plate

**Review by serenibyss — APPROVED** [review]
